backend/src/main/java/ai/verta/modeldb/reconcilers/UpdateParentTimestampReconcile.java

```diff
@@ -0,0 +1,92 @@
+package ai.verta.modeldb.reconcilers;
+
+import ai.verta.modeldb.common.reconcilers.ReconcileResult;
+import ai.verta.modeldb.common.reconcilers.Reconciler;
+import ai.verta.modeldb.common.reconcilers.ReconcilerConfig;
+import ai.verta.modeldb.utils.ModelDBHibernateUtil;
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+
+public class UpdateParentTimestampReconcile extends Reconciler<Long> {
+  private static final Logger LOGGER = LogManager.getLogger(UpdateParentTimestampReconcile.class);
+  private static final ModelDBHibernateUtil modelDBHibernateUtil =
+      ModelDBHibernateUtil.getInstance();
+
+  public UpdateParentTimestampReconcile(ReconcilerConfig config) {
+    super(config, LOGGER);
+  }
+
+  @Override
+  public void resync() {
+    var fetchUpdatedDatasetIds =
+        new StringBuilder("SELECT rc.repository_id ")
+            .append(" FROM commit cm INNER JOIN repository_commit rc ")
+            .append(" ON rc.commit_hash = cm.commit_hash ")
+            .append(" INNER JOIN commit_parent cp ")
+            .append(" ON cp.parent_hash IS NOT NULL ")
+            .append(" AND cp.child_hash = cm.commit_hash ")
+            .append(" INNER JOIN repository rp ")
+            .append(" ON rp.id = rc.repository_id AND rp.date_updated < cm.date_created ")
+            .append(" GROUP BY rc.repository_id")
+            .toString();
+
+    try (Session session = modelDBHibernateUtil.getSessionFactory().openSession()) {
+      Query fetchUpdatedDatasetIdQuery = session.createSQLQuery(fetchUpdatedDatasetIds);
+      fetchUpdatedDatasetIdQuery.setMaxResults(config.maxSync);
+      fetchUpdatedDatasetIdQuery.stream().forEach(id -> this.insert(Long.parseLong(id.toString())));
+    }
+  }
+
+  @Override
+  protected ReconcileResult reconcile(Set<Long> ids) {
+    LOGGER.debug("Reconciling projects " + ids.toString());
+
+    try (Session session = modelDBHibernateUtil.getSessionFactory().openSession()) {
```

Comment:
Can we just use jdbi? You're already not leveraging hibernate at all for this and we're trying to move away from hibernate.